UX-93 Storybook welcome page

phoenixeliot opened the pull request:
Initial draft, needs to be hooked in to actually display as the welcome page (probably as an MDX story?)

phoenixeliot committed:
Write first draft of welcome page

DanLipsitt commented on stories/Welcome-draft.md:
```diff
@@ -0,0 +1,45 @@
+
+How StoryBook renders Story Source blocks:
+- Source blocks show what React components are rendered, and their props (and any other HTML elements in the story). Other source code that goes into a story does not generally display in the Story Source panel.
+- When showing you the source code for a story, StoryBook displays function props as `noRefCheck`, rather than showing the function contents. When copying from these stories, you will have to provide actual functions here instead. In most cases, Ant is better reference for what functions are appropriate for these props.
```
This is fixed in the new DocsPage Storybook addon.

phoenixeliot replied:
I'll update this text when that gets merged, then

DanLipsitt commented on stories/Welcome-draft.md:
````diff
@@ -0,0 +1,45 @@
+const { MLGroup } = MLInput // or use MLInput.MLGroup directly, in your JSX
+```
+
+A few components must be imported directly from their path, or should be if you want tree-shaking (like with {linkTo('MLIcon')}):
````
How do we tell which components we need to do this with?

phoenixeliot replied:
I've updated it to just be about icons; the others are ones you don't need to be using directly anyway.

DanLipsitt commented on stories/Welcome-draft.md:
```diff
@@ -0,0 +1,45 @@
+See [Ant's page on customizing the theme](https://ant.design/docs/react/customize-theme) for more detail and a full list of theme variables provided by Ant.
+
+
+## How to use these components
```
I assume we don't need install info if people are getting the library from npm?

phoenixeliot replied:
I'm confused what you mean here, and which part of the code above you're referring to

DanLipsitt replied:
I thought I deleted this comment. Nothing to see here... 😊

DanLipsitt replied:
(ML said in our last call that we can assume that library install is documented elsewhere.)

phoenixeliot committed:
Merge branch 'develop' into 93_storybook-welcome

DanLipsitt commented on stories/Welcome-draft.md:
```diff
@@ -0,0 +1,45 @@
+
+TODO: Display this as a proper MDX file.
+
+# Welcome to the MarkLogic Design System Storybook
```
It might make sense to have a brief overview of what this storybook contains and what it is for.

phoenixeliot committed:
Improve copy based on CR

DanLipsitt reviewed (dismissed):
All good now. (Pending integration to display properly in the storybook.)

DanLipsitt commented:
@phoenixeliot Possibly another quirk for the gotchas list: icons will look different in the storybook code view depending on which icon set they’re from (I think?). For example, `<MLIcon.Route />` gets rendered as `<route />` and <MLIcon.SearchOutlined /> gets rendered as `<SearchOutlined />` . Things will work fine if they follow your instructions, but I’m not sure what will happen if they try to paste `<route />` , so you may want a specific warning about that. It might also be worth noting that following the Ant Icon docs may not work.

phoenixeliot committed:
Display welcome markdown as story

phoenixeliot committed:
Fix ant version in link

phoenixeliot committed:
Merge branch 'develop' into 93_storybook-welcome

phoenixeliot committed:
Updated information about using FontAwesome icons

phoenixeliot committed:
Fix fontawesome icon displayName

phoenixeliot commented:
I updated the text to reflect how FontAwesome icons should be accessed. I fixed the icon displayName capitalization in another branch (26_result), so I've moved that small fix here so the documentation is at least accompanied by the thing that makes it correct.

phoenixeliot committed:
Remove install notes; those go elsewhere

phoenixeliot committed:
Restrict welcome page CSS to that page

phoenixeliot committed:
Merge branch 'develop' into 93_storybook-welcome
